### PR TITLE
Non S3 URI as source resulting on local filesystem scanning

### PR DESCRIPTION
**Pull request by pranavmalik-msft**
### Description
An issue surfaced in Azcopy where users encountered a bug during copy operations with a source URI of "https://isd-storage.obs.ae-ad-1.g42cloud.com". This triggered a scan of the local file system, resulting in failures. Upon investigation, it was discovered that unsupported cloud platforms, such as **g42cloud**, were incorrectly identified as local sources by the backend.
As can be observed below-
f47954df-28a8-46ac-b3bb-fc5f388404d4

This PR addresses this issue by categorizing all unsupported cloud platforms as `Unknown `instead of `Local`, thus preventing repetitive scans of local files and ensuring early failure in processing.
According to the new flow it would be as follows:

### Implementation:
- Implemented a check to categorize unsupported cloud platforms as `Unknown`.

### Tests:
- Added Unit Tests to verify this flow and enhance code coverage.

### CODE REVIEW

File: cmd/validators.go [hunk 1]
```diff
@@ -164,6 +165,9 @@ func InferArgumentLocation(arg string) common.Location {
 			if common.IsGCPURL(*u) {
 				return common.ELocation.GCP()
 			}
+
+			// If none of the above conditions match, return Unknown
+			return common.ELocation.Unknown()
```
Comment (siminsavani-msft) [inline cmd/validators.go:170]: How does this impact non standard URLs?

**Review by siminsavani-msft — COMMENTED** [review]
Let's add some tests here!

File: cmd/validators.go (hunk 1 above)
Comment (pranavmalik-msft) [reply cmd/validators.go:170]: non standard URL wont be processed , they would follow the default case.

**Comment by adreed-msft** [comment]
Needs tests, otherwise lgtm

**Review by adreed-msft — CHANGES_REQUESTED** [review]
Needs tests

**Comment by pranavmalik-msft** [comment]
> Needs tests

Done

**Review by siminsavani-msft — APPROVED** [review]

**Review by adreed-msft — APPROVED** [review]

**Pull request merged**